File: 0889-buddy-strings/0889-buddy-strings.py
class Solution:
    def buddyStrings(self, s: str, goal: str) -> bool:
        f1 = [0] * 26
        f2 = [0] * 26
        for i in s: 
            f1[ord(i) - ord('a')] += 1
        for i in goal:
            f2[ord(i) - ord('a')] += 1
        if f1 != f2:
            return False
        cnt = 0
        two = False
        for i, j in zip(s, goal):
            if not i == j:
                cnt += 1
            if f1[ord(i) - ord('a')] >= 2:
                two = True
        if cnt == 0:
            return two
        return cnt == 2

File: 0889-buddy-strings/0889-buddy-strings.cpp
class Solution {
public:
    bool buddyStrings(string s, string goal) {
        vector<int> f1(26), f2(26);
        for(auto a : s) f1[a-'a']++;
        for(auto a : goal) f2[a-'a']++;
        if(f1 != f2) return 0;
        int cnt = 0, two = 0;
        for(int i = 0; i<s.size(); i++) {
            if(s[i] != goal[i]) cnt++;
            if(f1[s[i] - 'a'] >= 2) two = 1;
        }
        return cnt == 0 ? two : cnt == 2;
    }
};